File: Modules/Common/Resources/SIE.Web.Resources/Holidays/Commands/HolidayAddCommand.js
SIE.defineCommand('SIE.Web.Resources.Holidays.Commands.HolidayAddCommand', {
    extend: 'SIE.cmd.Add',
    meta: { text: "添加", group: "edit", iconCls: "iconfont icon-AddEntity icon-green" },
    canExecute: function (view) {
        return true;
    },
    onItemCreated: function (entity) {
        if (entity) {
            var model = entity.data;
            var me = this;
            this.view.execute({
                data: model,
                isSubmmit: false,
                success: function (res) {
                    var data = res.Result;
                    entity.setBeginDate(data.BeginDate)
                    entity.setEndDate(data.EndDate)
                }
            }, me.view);
        }
    }
});

File: Modules/Common/Resources/SIE.Web.Resources/Holidays/Commands/HolidayAddCommand.cs
using SIE.Resources.Holidays;
using SIE.Web.Command;
using System;

namespace SIE.Web.Resources.Holidays.Commands
{
    /// <summary>
    /// 法定假期添加命令
    /// </summary>
    public class HolidayAddCommand : ViewCommand
    {
        /// <summary>
        /// 法定假期添加命令
        /// </summary>
        /// <param name="args"></param>
        /// <param name="scope"></param>
        /// <returns></returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var holiday = args.Data.ToJsonObject<Holiday>();
            holiday.BeginDate = DateTime.Today;
            holiday.EndDate = DateTime.Today.AddDays(1);
            return holiday;
        }
    }
}
